src/go/rpk/pkg/cli/topic/describe.go

```diff
@@ -53,107 +66,116 @@ partitions section. By default, the summary and configs sections are printed.
-
-				tw := out.NewTable("KEY", "VALUE", "SOURCE")
-				defer tw.Flush()
-				types.Sort(resp)
-				for _, config := range resp.Resources[0].Configs {
-					var val string
-					if config.IsSensitive {
-						val = "(sensitive)"
-					} else if config.Value != nil {
-						val = *config.Value
+			for _, topic := range resp.Topics {
+				sections := out.NewMaybeHeaderSections(
+					out.ConditionalSectionHeaders(map[string]bool{
+						secSummary: summary,
+						secConfigs: configs,
+						secPart:    partitions,
+					})...,
+				)
+
+				sections.Add(secSummary, func() {
+					tw := out.NewTabWriter()
+					defer tw.Flush()
+					tw.PrintColumn("NAME", *topic.Topic)
+					if topic.IsInternal {
+						tw.PrintColumn("INTERNAL", topic.IsInternal)
 					}
-					tw.Print(config.Name, val, config.Source)
-				}
-			})
-
-			sections.Add(secPart, func() {
-				offsets := listStartEndOffsets(cl, topic, len(t.Partitions), stable)
-
-				tw := out.NewTable(describePartitionsHeaders(
-					t.Partitions,
-					offsets,
-				)...)
-				defer tw.Flush()
-				for _, row := range describePartitionsRows(
-					t.Partitions,
-					offsets,
-				) {
-					tw.Print(row...)
-				}
-			})
+					tw.PrintColumn("PARTITIONS", len(topic.Partitions))
+					if len(topic.Partitions) > 0 {
+						p0 := &topic.Partitions[0]
+						tw.PrintColumn("REPLICAS", len(p0.Replicas))
+					}
+					if err := kerr.ErrorForCode(topic.ErrorCode); err != nil {
+						tw.PrintColumn("ERROR", err)
+					}
+				})
+
+				sections.Add(secConfigs, func() {
+					req := kmsg.NewPtrDescribeConfigsRequest()
+					reqResource := kmsg.NewDescribeConfigsRequestResource()
+					reqResource.ResourceType = kmsg.ConfigResourceTypeTopic
+					reqResource.ResourceName = *topic.Topic
+					req.Resources = append(req.Resources, reqResource)
+
+					resp, err := req.RequestWith(context.Background(), cl)
+					out.MaybeDie(err, "unable to request configs: %v", err)
+					if len(resp.Resources) != 1 {
+						out.Die("config response returned %d resources when we asked for 1", len(resp.Resources))
+					}
+					err = kerr.ErrorForCode(resp.Resources[0].ErrorCode)
+					out.MaybeDie(err, "config response contained error: %v", err)
+
+					tw := out.NewTable("KEY", "VALUE", "SOURCE")
+					defer tw.Flush()
+					types.Sort(resp)
+					for _, config := range resp.Resources[0].Configs {
+						var val string
+						if config.IsSensitive {
+							val = "(sensitive)"
+						} else if config.Value != nil {
+							val = *config.Value
+						}
+						tw.Print(config.Name, val, config.Source)
+					}
+				})
+
+				sections.Add(secPart, func() {
+					offsets := listStartEndOffsets(cl, *topic.Topic, len(topic.Partitions), stable)
+
+					tw := out.NewTable(describePartitionsHeaders(
+						topic.Partitions,
+						offsets,
+					)...)
+					defer tw.Flush()
+					for _, row := range describePartitionsRows(
+						topic.Partitions,
+						offsets,
+					) {
+						tw.Print(row...)
+					}
+				})
+				fmt.Println()
```

Comment:
Can we conditionally remove the newline if it's the last element? (or conditionally add it if it's not the last haha) to avoid printing a newline for a single describe:

```
# before
$ rpk topic describe a -p
PARTITION  LEADER  EPOCH  REPLICAS  LOG-START-OFFSET  HIGH-WATERMARK
0          0       1      [0]       0                 7
$ rpk topic describe a -p
PARTITION  LEADER  EPOCH  REPLICAS  LOG-START-OFFSET  HIGH-WATERMARK
0          0       1      [0]       0                 7

$

Reply:
I think the test failure:

```
    obj["PARTITION"] = int(obj["PARTITION"])
ValueError: invalid literal for int() with base 10: ''
```

It is due to this. It is trying to parse the last row of the table, which is an empty line.

Reply:
Fixed.